### PR TITLE
DROOLS-3917 : The 'run all' panel should automatically close after leaving the project's library

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-3917

@kkufova @danielezonca @gitgabrio Here you are.

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/test/TestRunnerReportingScreen.java [hunk 1]
```diff
@@ -55,6 +63,12 @@ public Widget asWidget() {
         return panel.asWidget();
     }
 
+    public void onPlaceGainFocusEvent(final @Observes PlaceGainFocusEvent placeGainFocusEvent) {
+        if (!placeGainFocusEvent.getPlace().getIdentifier().equals(IDENTIFIER)) {
```
Comment (danielezonca) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/test/TestRunnerReportingScreen.java]: Are you sure `placeGainFocusEvent.getPlace().getIdentifier()` is always different from `null`?

Comment (Rikkola) [reply kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/test/TestRunnerReportingScreen.java]: Perhaps it is best to not trust any of these. I'll add checks. Place and identifier might be null.

Comment (Rikkola) [reply kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/test/TestRunnerReportingScreen.java]: PlaceRequest can't be null, but identifier could depending on what the implementation was.

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by Rikkola** [comment]
@gitgabrio The thing is, when do we know we left? This was the event that fits most cases.
For example I tried closing after we move to outside of Library perspective, but project list for example is still Library. I tried closing after the asset list is closed, but then opening an asset does not close it.

So I had a setup where the screen would hide when closing the asset list or when an asset is opened, but this breaks if you move to anywhere else, for example if we add settings screens or anything like that.

This now closes when ever focus changes and it does have problems, but any screen that would get focus is likely to close this screen.

**Comment by gitgabrio** [comment]
@Rikkola 
Hi, I understand that was the underlying issue. Moreover, I do not know exactly the context where this happens. Just an idea (may be wrong, of course): if "project library" is a widget/view/screen/whatever, moving to another widget *should* fire a **lost** focus event for it. So, would not possible to intercept and react on this ?

**Comment by Rikkola** [comment]
@gitgabrio Yes, but the library loses the focus when the test results gains it. And the test results only loses the focus if the user moves out of the list, either to project list or into an asset.

**Comment by gitgabrio** [comment]
@Rikkola 
OK, ok, no way, you know better! :smile:

**Comment by Rikkola** [comment]
@gitgabrio Not necessary, if there are better ways I'm open to change, but due to the workbench structure I can't think of any better ways.

**Comment by kkufova** [comment]
@gitgabrio, could you please approve this?

**Pull request merged**